report/string_set.go

```diff
@@ -81,32 +81,49 @@ func (s StringSet) Add(strs ...string) StringSet {
 }
 
 // Merge combines the two StringSets and returns a new result.
-func (s StringSet) Merge(other StringSet) StringSet {
+// Second return value is true if the return value is s
+func (s StringSet) Merge(other StringSet) (StringSet, bool) {
 	switch {
 	case len(other) <= 0: // Optimise special case, to avoid allocating
-		return s // (note unit test DeepEquals breaks if we don't do this)
+		return s, true // (note unit test DeepEquals breaks if we don't do this)
 	case len(s) <= 0:
-		return other
+		return other, false
 	}
-	result := make(StringSet, len(s)+len(other))
-	for i, j, k := 0, 0, 0; ; k++ {
+
+	i, j := 0, 0
+loop:
+	for i < len(s) {
 		switch {
-		case i >= len(s):
-			copy(result[k:], other[j:])
-			return result[:k+len(other)-j]
-		case j >= len(other):
-			copy(result[k:], s[i:])
-			return result[:k+len(s)-i]
-		case s[i] < other[j]:
-			result[k] = s[i]
+		case j >= len(other) || s[i] < other[j]:
+			i++
+		case s[i] == other[j]:
 			i++
-		case s[i] > other[j]:
-			result[k] = other[j]
 			j++
-		default: // equal
-			result[k] = s[i]
+		default:
+			break loop
+		}
+	}
+	if i >= len(s) && j >= len(other) {
+		return s, true
+	}
+
+	result := make(StringSet, i, len(s)+len(other))
+	copy(result, s[:i])
+
+	for i < len(s) {
+		switch {
+		case j >= len(other) || s[i] < other[j]:
```

Comment:
why not split these two cases, as we used to, which gets us a bulk append and early loop termination for the first case?

Reply:
The simple answer is that I copy-pasted the code from `LatestMap`.  It seems the same improvement would work there.